### PR TITLE
Use medium type to determine whether we mount a path as ramdisk or not

### PR DESCRIPTION
**Pull request by yuzhu**
behavior before this change: if we have multiple path on tier 0, mounting will fail
behavior after this change: if we have multiple path on tier 0, we will mount any path that has a medium type of "MEM"

### CODE REVIEW

**Comment by yuzhu** [comment]
@madanadit PTAL, this one addresses the issue when we have multiple path in the first tier.

File: bin/alluxio-mount.sh [hunk 1]
```diff
@@ -31,6 +31,17 @@ function init_env() {
...
+  IFS=','
+  read -ra PATHARRAY <<< "$TIER_PATH"
+  read -ra MEDIUMTYPEARRAY <<< "$MEDIUM_TYPE"
+  RAMDISKARRAY=()
+  for i in "${!PATHARRAY[@]}"; do 
+    if [ "${MEDIUMTYPEARRAY[$i]}" = "MEM" ]; then
+      RAMDISKARRAY+=(${PATHARRAY[$i]})
+    fi
+  done
+  IFS=' '
```
Comment (madanadit) [inline bin/alluxio-mount.sh]: nit: store old IFS before l35 and restore here

**Comment by yuzhu** [comment]
jenkins, test this please

File: bin/alluxio-mount.sh [hunk 2]
```diff
@@ -31,6 +31,17 @@ function init_env() {
   MEM_SIZE=$(${BIN}/alluxio getConf --unit B alluxio.worker.memory.size)
   TIER_ALIAS=$(${BIN}/alluxio getConf alluxio.worker.tieredstore.level0.alias)
   TIER_PATH=$(${BIN}/alluxio getConf alluxio.worker.tieredstore.level0.dirs.path)
```
Comment (madanadit) [inline bin/alluxio-mount.sh]: nit: rename to TIER_PATHS

File: bin/alluxio-mount.sh [hunk 3]
```diff
@@ -43,48 +54,48 @@ function check_space_linux() {
 }
 
 function mount_ramfs_linux() {
```
Comment (madanadit) [inline bin/alluxio-mount.sh]: nit: TIER_PATH=${1} and rest remains unchanged

File: bin/alluxio-mount.sh [hunk 4]
```diff
@@ -43,48 +54,48 @@ function check_space_linux() {
...
+    chmod a+w ${1}
   fi
   if [[ $? -ne 0 ]]; then
-    echo "ERROR: chmod RamFS ${TIER_PATH} failed" >&2
+    echo "ERROR: chmod RamFS ${1} failed" >&2
     exit 1
   fi
 }
 
 function umount_ramfs_linux() {
```
Comment (madanadit) [inline bin/alluxio-mount.sh]: nit: same here. TIER_PATH=${1}

File: bin/alluxio-mount.sh [hunk 5]
```diff
@@ -95,39 +106,53 @@ function mount_ramfs_mac() {
...
 
   # Format the RAM FS
-  echo "Formatting RamFS: ${TIER_PATH} ${num_sectors} sectors (${MEM_SIZE})."
+  echo "Formatting RamFS: ${1} ${num_sectors} sectors (${MEM_SIZE})."
   # Remove the "/Volumes/" part so we can get the name of the volume.
-  diskutil erasevolume HFS+ ${TIER_PATH/#\/Volumes\//} $(hdiutil attach -nomount ram://${num_sectors})
+  diskutil erasevolume HFS+ ${1/#\/Volumes\//} $(hdiutil attach -nomount ram://${num_sectors})
 }
 
 function umount_ramfs_mac() {
```
Comment (madanadit) [inline bin/alluxio-mount.sh]: same here

File: bin/alluxio-mount.sh [hunk 6]
```diff
@@ -95,39 +106,53 @@ function mount_ramfs_mac() {
...
+    echo "Unmounting ramfs at ${1}"
     hdiutil detach -force ${device}
   else
-    echo "Ramfs is not currently mounted at ${TIER_PATH}"
+    echo "Ramfs is not currently mounted at ${1}"
   fi
 }
 
+function mount_ramfs_local_all() {
+for i in "${RAMDISKARRAY[@]}"
```
Comment (madanadit) [inline bin/alluxio-mount.sh]: indent

File: bin/alluxio-mount.sh [hunk 7]
```diff
@@ -95,39 +106,53 @@ function mount_ramfs_mac() {
...
+do 
+  mount_ramfs_local $i
+done
+}
+
 function mount_ramfs_local() {
```
Comment (madanadit) [inline bin/alluxio-mount.sh]: same here

File: bin/alluxio-mount.sh [hunk 8]
```diff
@@ -95,39 +106,53 @@ function mount_ramfs_mac() {
...
     check_space_linux
-    umount_ramfs_linux
-    mount_ramfs_linux
+    umount_ramfs_linux $1
+    mount_ramfs_linux $1
   fi
 }
 
+function umount_ramfs_local_all() {
+for i in "${RAMDISKARRAY[@]}"
```
Comment (madanadit) [inline bin/alluxio-mount.sh]: indent

File: bin/alluxio-mount.sh [hunk 9]
```diff
@@ -95,39 +106,53 @@ function mount_ramfs_mac() {
...
+do 
+  umount_ramfs_local $i
+done
+}
+
 function umount_ramfs_local() {
   if [[ $(uname -a) == Darwin* ]]; then
-    umount_ramfs_mac
+    umount_ramfs_mac $1
```
Comment (madanadit) [inline bin/alluxio-mount.sh]: same here

Comment (madanadit) [reply bin/alluxio-mount.sh]: ^

Comment (yuzhu) [reply bin/alluxio-mount.sh]: i think it is ok to leave this function parameterized right?

File: bin/alluxio-mount.sh (hunk 6 above)
Comment (yuzhu) [reply bin/alluxio-mount.sh]: fixed indent

File: bin/alluxio-mount.sh (hunk 9 above)
Comment (madanadit) [reply bin/alluxio-mount.sh]: okay, your call

File: bin/alluxio-mount.sh [hunk 10]
```diff
@@ -110,24 +126,38 @@ function umount_ramfs_mac() {
   fi
 }
 
+function mount_ramfs_local_all() {
+  for i in "${RAMDISKARRAY[@]}"
+  do
+    mount_ramfs_local $i
```
Comment (calvinjia) [inline bin/alluxio-mount.sh]: Is it conventional to use `i` here?

Comment (yuzhu) [reply bin/alluxio-mount.sh]: changed to a more descriptive var name

**Comment by yuzhu** [comment]
@calvinjia Not only that, we also force the user to have "MEM" as the tier0 tier name, otherwise we won't mount.  we may need to update our documentation to match this behavior.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**